docs/DeveloperGuide.adoc

```diff
@@ -311,14 +329,146 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+. The input given by the user is passed to `ArgumentTokeniser#tokenise` to split the input separated by prefixes.
+. This returns a `ArgumentMultiMap` which contains a map with prefixes as keys and their associated input arguments as the value.
+. The string associated with `d/` is then passed into `ListCommandParser#splitByWhitespace` for further processing and returns an array.
+. The argument mode is determined by the size of this array and the elements are further processed into `Date` objects, before creating and returning a `ListCommand` object.
+
+The `ListCommand` has two constructors which makes use of overloading to reduce code complexity.
+
+* One constructor has no arguments and assigns default predicate for the `FilteredList` in `ModelManager`,
+`PREDICATE_SHOW_ALL_RECORDS` which will show all items in the list.
+* The second constructor takes in 2 `Date` arguments and assigns the predicate `DateIsWithinDateIntervalPredicate` which will only show items within the date interval.
+
+The following sequence diagram shows how the list operation works:
+
+image::ListSequenceDiagram.png[width="800"]
+
+For simplicity, interactions with the UI is not shown in the diagram above. +
+The update of the UI `RecordListPanel` is done through the event system. `FilteredList` is a type of ObservableList
+implemented by the Java 8 API and it will propagate any changes to the list to any listeners listening to it. This
+listener is present in `RecordListPanel` and will update the UI list automatically.
+
+=== Find records by tag feature
+==== Current Implementation
+The findtag mechanism is also facilitated by `ModelManager`. FindTagCommand calls `ModelManager#updateFilteredRecords`
+and passes in different predicates depending on the input by the user.
+
+This feature has only one keyword `findtag` and a single working mode which takes in any number of input arguments. The
+input given by the user is passed to `FindTagCommandParser#parse` to split the desired tags the user wants to search by
+into an array of strings. The array of strings is passed into `TagsContainsKeywordsPredicate` to create the predicate
+for `updateFilteredRecordList` required in `ModelManager`.
+
+In `TagsContainsKeywordsPredicate`, to compare for a match, every keyword in the array is compared
+against the set of tags of each record and as long as any tag matches any of the keywords,
+the predicate will evaluate to true and allows the `FilteredList` to filter out the records that do not fulfil the
+predicate.
+
+`FindTagCommandParser` returns a `FindTagCommand` object which calls `updateFilteredRecordList` to set the new predicate
+and obtain a new `filteredRecords` based on the predicate, which will also trigger an event for the UI to read in and display the new records.
+
+The following sequence diagram shows how the limit operation works:
+
+image::FindTagSequenceDiagram.png[width:800]
```

Comment:
Very good explanation and good job on illustrating the new feature with the sequence diagram